### PR TITLE
Review request

### PR DESCRIPTION
**Pull request by Zamtakaish**

### CODE REVIEW

**Review by roman-karpovich — COMMENTED** [review]
hey

File: README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Simple RESTfull app
```
Comment (pgutnik) [inline README.md:1]: ping

File: src/main/java/com/mastery/java/task/dto/Employee.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                "employeeId=" + employeeId +
+                ", firstName='" + firstName + '\'' +
+                ", lastName='" + lastName + '\'' +
+                ", departmentId=" + departmentId +
+                ", jobTitle='" + jobTitle + '\'' +
+                ", gender=" + gender +
+                ", dateOfBirth='" + dateOfBirth + '\'' +
+                '}';
+    }
+}
```
Comment (pgutnik) [inline src/main/java/com/mastery/java/task/dto/Employee.java:96]: Add equals/hashcode overrides.
https://www.baeldung.com/java-equals-hashcode-contracts

File: src/main/java/com/mastery/java/task/dao/EmployeeDao.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        isEmployeeExist(id);
+        return repository.save(employee);
+    }
+
+    public void deleteEmployee(int id){
+        isEmployeeExist(id);
+        repository.deleteById(id);
+    }
+
+    private void isEmployeeExist(int id) {
```
Comment (pgutnik) [inline src/main/java/com/mastery/java/task/dao/EmployeeDao.java:44]: Seems a bit redundant. Maybe deleteById(int) will throw it's own exception that you can rethrow up?

File: src/main/java/com/mastery/java/task/dao/EmployeeDao.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Autowired
+    private EmployeeRepository repository;
+
+    public List<Employee> getAllEmployees() {
+        return repository.findAll();
+    }
+
+    public Employee getEmployee(int id) {
+        Optional<Employee> employee = repository.findById(id);
```
Comment (pgutnik) [inline src/main/java/com/mastery/java/task/dao/EmployeeDao.java:23]: Can be replaced by isEmployeeExists.

File: src/main/java/com/mastery/java/task/dto/Employee.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    private Short departmentId;
+    private String jobTitle;
+    @Enumerated(EnumType.STRING)
+    private Gender gender;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+    private Date dateOfBirth;
+
+    public Employee() {
+    }
+
```
Comment (pgutnik) [inline src/main/java/com/mastery/java/task/dto/Employee.java:27]: Maybe add a constructor with all fields?

File: src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        testEmployee.setFirstName("Test");
+        testEmployee.setLastName("Test");
+        testEmployee.setDepartmentId((short) 2);
+        testEmployee.setJobTitle("Test");
+        testEmployee.setGender(Gender.Male);
+        testEmployee.setDateOfBirth(new Date());
+    }
+
+    @Test
+    public void shouldListAllUsers_whenGetMethodCalled() throws Exception {
```
Comment (pgutnik) [inline src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java:55]: No need to use "_" just write long names, welcome to Java naming.

File: src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private EmployeeService serviceMock;
+
+    private final Employee testEmployee = new Employee();
+
+    public void setTestEmployee(){
```
Comment (pgutnik) [inline src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java:44]: createTestEmployee is better

File: src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    @Test
+    public void shouldThrowException_ifEmployeeNotFound() throws Exception {
+
+        setTestEmployee();
+
+        int id = testEmployee.getEmployeeId();
+
+        Mockito.doThrow(new ResponseStatusException(HttpStatus.NOT_FOUND, "Employee is not found with ID: " + id))
+                .when(serviceMock).getEmployee(id);
+        mvc.perform(get("/employees/" + id)
```
Comment (pgutnik) [inline src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java:94]: Add a new line to separate When from Then parts of the test.

File: src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    @Test
+    public void shouldCreateUser_whenPostMethodCalled() throws Exception {
+
+
+
+        when(serviceMock.createEmployee((ArgumentMatchers.any(Employee.class)))).thenReturn(testEmployee);
+
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (pgutnik) [inline src/test/java/com/mastery/java/task/controller/EmployeeControllerTest.java:106]: Can this mapper be extracted to a test class field?

File: src/test/java/com/mastery/java/task/dao/EmployeeDaoTest.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+public class EmployeeDaoTest {
+
+    @Mock
+    private EmployeeRepository repositoryMock;
+    @InjectMocks
+    private EmployeeDao employeeDao;
+
+    private final Employee testEmployee = new Employee();
+
+    private void setTestEmployee(){
```
Comment (pgutnik) [inline src/test/java/com/mastery/java/task/dao/EmployeeDaoTest.java:31]: Rename to createTestEmployee()

File: src/test/java/com/mastery/java/task/dao/EmployeeDaoTest.java [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+        testEmployee.setDateOfBirth(new Date());
+    }
+
+
+    @Test
+    public void shouldReturnAllEmployees(){
+        setTestEmployee();
+
+        List<Employee> employeeList = new ArrayList();
+        employeeList.add(testEmployee);
```
Comment (pgutnik) [inline src/test/java/com/mastery/java/task/dao/EmployeeDaoTest.java:47]: Replace with Arrays.asList()

**Review by pgutnik — CHANGES_REQUESTED** [review]
Looks good, but some changes are required.

**Pull request merged**